tests/integration_tests/security/guest_token_security_tests.py

```diff
@@ -195,3 +204,238 @@ def test_raise_for_access_dashboard_as_guest_no_rbac(self):
 
         db.session.delete(dash)
         db.session.commit()
+
+
+@patch.dict(
+    "superset.extensions.feature_flag_manager._feature_flags",
+    EMBEDDED_SUPERSET=True,
+)
+@pytest.mark.usefixtures(
+    "create_dataset",
+    "load_birth_names_dashboard_with_slices_class_scope",
+    "load_world_bank_dashboard_with_slices_class_scope",
+)
+class TestGuestUserDatasourceAccess(SupersetTestCase):
+    """
+    Guest users should only have access to datasources that are associated with a
+    dashboard they have access to, and only with that dashboard context present
+    """
+
+    @pytest.fixture(scope="class")
+    def create_dataset(self):
+        with self.create_app().app_context():
+            dataset = SqlaTable(
+                table_name="dummy_sql_table",
+                database=get_example_database(),
+                schema=get_example_default_schema(),
+                sql="select 123 as intcol, 'abc' as strcol",
+            )
+            session = db.session
+            session.add(dataset)
+            session.commit()
```

Comment:
I suspect the commit is unnecessary. This would mean that lines 240 and 241 become.

```
db.session.rollback()
```

Reply:
Yeah I guess it's unnecessary for these and we could instead just run these tests within an uncommitted transaction.  Do you feel that's preferable? Here I pretty much just copied the pattern from existing fixtures I saw which all committed their changes.